Classes/TextUtils.cs

```diff
@@ -24,19 +24,106 @@ namespace RadioDld
+        private static string reDaySuffix = "(?<=[0-9])(?:st |nd |rd |th )";
+        private static string reMonth = @"(2\d|1\d|0?\d)"; // m(mm)
+        private static string reMonthText = @"(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:t(?:ember)?)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)"; // (mmm)
+        private static string reYear = @"((?:(?:\d{4})|(?:\d{2})(?![\\d])|(?:\')(?:\d{2})(?![\\d])))"; // yyyy('yy)(yy)
+        private static string reWS = @"( ?)"; // trailing white space
+
+        // Key date formats to look out for
+        // 1 / d(dd)(st | nd | rd | th) | m(mm)(mmm) | yyyy('yy)(yy)
+        private static string reF1 = @reDay + reDelim + "(" + reMonth + "|" + reMonthText + ")" + reDelim + reYear + reWS;
+
+        // 2 / yyyy('yy)(yy)|m(mm)(mmm)|d(dd)(st|nd|rd|th)
+        private static string reF2 = @"(" + reYear + reDelim + "(" + reMonth + "|" + reMonthText + ")" + reDelim + reDay + reWS + ")";
+
+        // 3 / mmm|d(dd)(st|nd|rd|th)|yyyy('yy)(yy)
+        private static string reF3 = @"(" + reMonthText + ")" + reDelim + reDay + reDelim + reYear + reWS;
+
+        // Build RegEx string
+        private static Regex matchStripDate = new Regex("(" + reF1 + "|" + reF2 + "|" + reF3 + ")", RegexOptions.IgnoreCase | RegexOptions.ECMAScript);
+        private static Regex removeDaySuffix = new Regex(reDaySuffix, RegexOptions.IgnoreCase);
+
+        // Set day range for Similar date check
+        private static int iSimilarDateDayRange = 6;
+        //// private static int iSimilarDateDayRange = 3;
+
+        /* OLD FUNCTION for record
+        *  public static string oldStripDateFromName(string name, DateTime stripDate)
+        *  {
+        *      // Use regex to remove a number of different date formats from episode titles.
+        *      // Will only remove dates with the same month & year as the programme itself, but any day of the month
+        *      // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
+        *      Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+        *      if (matchStripDate.IsMatch(name))
+        *      {
+        *          name = matchStripDate.Match(name).Groups["name"].ToString();
+        *      }
+        *      return name;
+        *  }
+        */
```

Comment:
This isn't necessary as it can be seen in the revision history.

Reply:
Agree, kept it really jsut i so had it handy when trying to work out what the current regex was doing. I'll remove this as i clean up the other comments etc.